src/frequenz/sdk/actor/power_distributing/_component_pool_status_tracker.py

```diff
@@ -0,0 +1,163 @@
+        self._merged_status_receiver = self._make_merged_status_receiver()
+
+        self._task = asyncio.create_task(self._run())
+
+    async def join(self) -> None:
+        """Wait and return when the instance's task completes.
+
+        It will not terminate the instance, which can be done with the `stop` method.
+        """
+        await self._task
+
+    async def stop(self) -> None:
+        """Stop tracking batteries status."""
+        await cancel_and_await(self._task)
+        await asyncio.gather(
+            *[tracker.stop() for tracker in self._component_status_trackers],
+        )
+        await self._merged_status_receiver.stop()
+
+    def _make_merged_status_receiver(
+        self,
+    ) -> Merge[ComponentStatus]:
+        status_receivers: list[Receiver[ComponentStatus]] = []
+
+        for component_id in self._component_ids:
+            channel: Broadcast[ComponentStatus] = Broadcast(
+                f"component_{component_id}_status"
+            )
+            tracker = self._component_status_tracker_type(
+                component_id=component_id,
+                max_data_age_sec=self._max_data_age_sec,
+                max_blocking_duration_sec=self._max_blocking_duration_sec,
+                status_sender=channel.new_sender(),
+                set_power_result_receiver=self._set_power_result_channel.new_receiver(),
+            )
+            self._component_status_trackers.append(tracker)
+            status_receivers.append(channel.new_receiver())
+        return Merge(*status_receivers)
+
+    async def _run(self) -> None:
+        """Start tracking component status."""
+        while True:
+            try:
+                await self._update_status()
+            except Exception as err:  # pylint: disable=broad-except
+                _logger.error(
+                    "ComponentPoolStatus failed with error: %s. Restarting.", err
+                )
+
+    async def _update_status(self) -> None:
+        async for status in self._merged_status_receiver:
+            component_id = status.component_id
+            if status.value == ComponentStatusEnum.WORKING:
+                self._current_status.working.add(component_id)
+                self._current_status.uncertain.discard(component_id)
+            elif status.value == ComponentStatusEnum.UNCERTAIN:
+                self._current_status.working.discard(component_id)
+                self._current_status.uncertain.add(component_id)
+            elif status.value == ComponentStatusEnum.NOT_WORKING:
+                self._current_status.working.discard(component_id)
+                self._current_status.uncertain.discard(component_id)
+
+            await self._component_status_sender.send(self._current_status)
+
+    async def update_status(
+        self, succeed_components: set[int], failed_components: set[int]
+    ) -> None:
+        """Notify which components succeed and failed in the request.
```

Comment:
I think this is mixing present and past for succeed and failed, respectively. If this comment applies, then argument names need to be updated too

Reply:
I'll fix this particular case, but I'm sure there are other instances as well.  This is old code, and this PR has a limited scope.

Reply:
Thanks!

Reply:
done here: https://github.com/frequenz-floss/frequenz-sdk-python/commit/61765a81